### ISSUE TITLE
Command line for Thread titme (with Tasks) Stacks

### ISSUE THREAD

**Issue by ashishnegi** [microsoft/perfview#954]
From command line, if I use : 

> PerfView.exe /BufferSizeMB:1024 /InMemoryCircularBuffer /zip:false /merge:false /circularmb:2048 /nogui /ThreadTime collect perfdetails.etl

And then if I open perfdetails.etl in PerfView, I don’t see `Thread time (with Tasks) Stacks`. I only see `Thread Time Stacks`.
 
What is the command line parameter for generating etl that can show `Tasks stack` in PerfView ?

**Comment by windhandel** [comment]
If, like me, you were looking for the Thread Time checkbox in the UI, it's waaaaayyyy out of sequence. It's near the top. The switch is /threadTime.

See:

![image](https://user-images.githubusercontent.com/5337357/124983237-8dd8be80-dfec-11eb-9223-68bd76e7a39d.png)

And:

![image](https://user-images.githubusercontent.com/5337357/124983280-9b8e4400-dfec-11eb-9311-c9b65a112960.png)

**Comment by kirsan31** [comment]
Same problem here `Thread time (with Tasks) Stacks` is missing, only `Thread Time Stacks`...

**Comment by brianrob** [comment]
@kirsan31, you will also need to enable the TPL events.  These are not enabled by default as they are quite verbose and can impact CPU sampling results.

**Comment by kirsan31** [comment]
@brianrob 
> you will also need to enable the TPL events. These are not enabled by default as they are quite verbose and can impact CPU sampling results.

Yes, sure:
![image](https://user-images.githubusercontent.com/17767561/161325502-486915aa-dd8d-4aac-9a73-2a4b7c77f92f.png)
![image](https://user-images.githubusercontent.com/17767561/161325648-bf905996-e91f-45f3-94bb-deab41ef7ea6.png)

**Comment by brianrob** [comment]
Interesting.  Are you by chance able to share the trace?  It looks like this view should show up as long as you specify /ThreadTime and TPL events.

**Comment by kirsan31** [comment]
@brianrob do you need full trace or only log? If full - can I some how send it to you in private manner?

**Comment by brianrob** [comment]
Yes, I will need the full trace.  Feel free to post it to a site such as OneDrive and then send a link to me at brianrob [at] microsoft.

**Comment by kirsan31** [comment]
> Yes, I will need the full trace. Feel free to post it to a site such as OneDrive and then send a link to me at brianrob [at] microsoft.

Sent from golikova at gmail.

**Comment by brianrob** [comment]
Got it. Thanks.

The issue is that you are capturing the TPL events, but it appears that stacks are not being collected, and this view requires stacks on these events in order to be useful.  When I test this on my local machine, I do get the views that you're expecting:

<img width="259" alt="image" src="https://user-images.githubusercontent.com/6210322/162036693-07fff7bb-589b-4571-850f-149d4f00225c.png">

Looking at the code, we filter the set of stacks to enable down to just TaskScheduled, TaskWaitSend, and AwaitTaskContinuationScheduled to avoid additional stack walking overhead since this provider is quite verbose.  This is done by a Win8+ API that we check for by checking the version number of `%Windir%\system32\Kernel32.dll`.  Given that you captured this trace on Win 10, I have to believe that something is going wrong with this check - perhaps PerfView isn't properly opening the file, or something like that.

One thing that you can try is rather than specifying `/TPLEvents:Default`, specify `/Providers:.NETTasks:0x1ff:@StacksEnabled=true`.  This will turn on stacks for all of the TPL events, and will bypass the check that seems to be failing on this machine.

**Comment by kirsan31** [comment]
Thank you!

> This is done by a Win8+ API that we check for by checking the version number of `%Windir%\system32\Kernel32.dll`. Given that you captured this trace on Win 10, I have to believe that something is going wrong with this check - perhaps PerfView isn't properly opening the file, or something like that.

Maybe this is due to the fact that we have a _rare_ version of the OS - `win 10 for workstation`?

**Comment by brianrob** [comment]
That's a good question - I don't know, and I'm not even quite sure where to get a copy to check.

**Comment by kirsan31** [comment]
> That's a good question - I don't know, and I'm not even quite sure where to get a copy to check.

I think I can do some experiments, if you interested and have any ideas about this...

**Comment by brianrob** [comment]
Yes, if you're able, it would be awesome to see what happens [here](https://github.com/microsoft/perfview/blob/main/src/PerfView/CommandProcessor.cs#L711) in a debugger.  This line ends up opening `kernel32.dll` and that's where I think things are going wrong on your machine.  If you're able to build a copy of PerfView and debug it, that would be great.

**Comment by kirsan31** [comment]
@brianrob 
I've build a version from [perfview-3.0](https://github.com/microsoft/perfview/tree/feature/perfview-3.0) branch (because I have 
 only vs2022) with extra debug logging.
Result:
`winDir C:\WINDOWS; kernel32 C:\WINDOWS\system32\Kernel32.dll`
`versionInfo 10.0.19041.1503 (WinBuild.160101.0800)`
check have passed and `Thread time (with Tasks) Stacks` is present.
Then I reverted to def. version and collected traces several times - `Thread time (with Tasks) Stacks` was always present.
So this is intermittent problem 🤷‍♂️

**Comment by brianrob** [comment]
Interesting.  That one I can't explain...

**Comment by brianrob** [comment]
Closing as not repro.
